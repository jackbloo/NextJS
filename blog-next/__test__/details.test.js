import { render, screen } from '@testing-library/react'
import { act } from 'react-dom/test-utils'
import Details from '../pages/blogs/[id]'

describe('Unit Testing of Details', () => {
  it('Should renders details component', async () => {
      await act(async () =>{
        await render(<Details data={[{name: 'test', image: 'https://google.com', content: 'test'}]}/>)
      })

    const heading = screen.getByTestId('detail-container')
    expect(heading).toMatchSnapshot();
  })
})