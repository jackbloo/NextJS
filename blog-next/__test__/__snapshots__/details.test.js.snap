// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit Testing of Details Should renders details component 1`] = `
<div
  class="container"
  data-testid="detail-container"
>
  <main
    class="main"
  >
    <div
      class="navbar"
      data-testid="navbar-container"
    >
      <div
        class="rightNavbar"
      >
        <a
          href="/"
        >
          Digitsense Blog
        </a>
      </div>
      <div
        class="leftNavbar"
      >
        <div
          class="nav"
          style="font-weight: normal; text-decoration: none;"
        >
          <a
            href="/"
          >
            Home
          </a>
        </div>
        <div
          class="nav"
          style="font-weight: normal; text-decoration: none;"
        >
          <a
            href="/about"
          >
            About us
          </a>
        </div>
        <div
          class="nav"
          style="font-weight: normal; text-decoration: none;"
        >
          <a
            href="/undefined"
          />
        </div>
      </div>
    </div>
    <h1
      class="title"
    >
      test
    </h1>
    <div
      class="picture"
    >
      <span
        style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
      >
        <span
          style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
        >
          <img
            alt=""
            aria-hidden="true"
            src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27800%27%20height=%27400%27/%3e"
            style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
          />
        </span>
        <img
          alt="Minimalist"
          data-nimg="intrinsic"
          decoding="async"
          src="/_next/image?url=https%3A%2F%2Fgoogle.com&w=1920&q=75"
          srcset="/_next/image?url=https%3A%2F%2Fgoogle.com&w=828&q=75 1x, /_next/image?url=https%3A%2F%2Fgoogle.com&w=1920&q=75 2x"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
        />
        <noscript />
      </span>
    </div>
    <div
      style="width: 50%; text-align: center;"
    >
      <p
        class="text"
      >
        test
      </p>
    </div>
  </main>
  <footer
    class="footer"
    data-testid="footer-container"
  >
    <a
      href="https://digit-sense.com/"
      rel="noopener noreferrer"
      target="_blank"
    >
      Powered by
       
      <span
        class="logo"
      >
        <span
          style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
        >
          <span
            style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2772%27%20height=%2740%27/%3e"
              style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
            />
          </span>
          <img
            alt="Vercel Logo"
            data-nimg="intrinsic"
            decoding="async"
            src="/_next/image?url=%2Fdigitsense.jpeg&w=256&q=75"
            srcset="/_next/image?url=%2Fdigitsense.jpeg&w=96&q=75 1x, /_next/image?url=%2Fdigitsense.jpeg&w=256&q=75 2x"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
          <noscript />
        </span>
      </span>
    </a>
  </footer>
</div>
`;
